packages/browser/src/transports/fetch.ts

```diff
@@ -12,19 +12,31 @@ export class FetchTransport extends BaseTransport {
    * @inheritDoc
    */
   public sendEvent(event: Event): PromiseLike<Response> {
-    const eventType = event.type || 'event';
+    return this._sendRequest(eventToSentryRequest(event, this._api), event);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public sendSession(session: Session): PromiseLike<Response> {
+    return this._sendRequest(sessionToSentryRequest(session, this._api), session);
+  }
 
-    if (this._isRateLimited(eventType)) {
+  /**
+   * @param sentryRequest Prepared SentryRequest to be delivered
+   * @param event original payload used to create SentryRequest
+   */
+  private _sendRequest(sentryRequest: SentryRequest, event: Event | Session): PromiseLike<Response> {
+    if (this._isRateLimited(sentryRequest.type)) {
       return Promise.reject({
         event,
```

Comment:
It is a bit strange that this promise returns an object that is probably different in shape than what a successful response from the server would resolve to?!

Not sure about the suggestion here, as it would change the type of the object.

```suggestion
        originalPayload,
```

Alternatively, if the original payload is an event include the `event`, otherwise include a new key `session` with a session.

Yet another alternative, breaking change, remove the event from the returned value altogether, then we don't need the `originalPayload` argument.

Reply:
Agree for sure, but I didn't want to break the existing behavior, as someone might rely on it, for example, to store events offline for later retries. Especially that in the end, they are all just envelopes.
I added an optional `type` key to allow for easier distinction for now.